File: engine-desktop/src/com/magma/engine/debug/Crasher.java
package com.magma.engine.debug;

import java.awt.Dimension;
import java.io.IOException;

import javax.swing.JEditorPane;
import javax.swing.JFrame;
import javax.swing.JLabel;

public class Crasher extends JFrame {

    private Crasher(String htmlPath) {
        super("Whoops, something went horribly wrong!");

        setResizable(true);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        Dimension size = new Dimension(600, 600);
        setPreferredSize(size);
        setMinimumSize(size);

        try {
            JEditorPane editor = new JEditorPane("file://"+htmlPath);
            editor.setEditable(false);
            getContentPane().add(editor);
            pack();

        } catch (IOException ec) {
            JLabel label = new JLabel("Could not display content.");
            getContentPane().add(label);
            ec.printStackTrace();
        }
        setVisible(true);
    }

    public static Crasher showCrashDialog(Exception e) {
        MagmaLogger.logException(e);
        String path = MagmaLoggerDesktop.flush(true);
        return new Crasher(path);
    }
}

File: engine/src/com/magma/engine/debug/Crasher.kt
package com.magma.engine.debug

import com.magma.engine.debug.MagmaLogger.logException
import java.awt.Dimension
import java.io.IOException
import javax.swing.JEditorPane
import javax.swing.JFrame
import javax.swing.JLabel

class Crasher private constructor(htmlPath: String) : JFrame("Whoops, something went horribly wrong!") {
    init {
        isResizable = true
        defaultCloseOperation = EXIT_ON_CLOSE
        val size = Dimension(600, 600)
        preferredSize = size
        minimumSize = size
        try {
            val editor = JEditorPane("file://$htmlPath")
            editor.isEditable = false
            contentPane.add(editor)
            pack()
        } catch (ec: IOException) {
            val label = JLabel("Could not display content.")
            contentPane.add(label)
            ec.printStackTrace()
        }
        isVisible = true
    }

    companion object {
        fun showCrashDialog(e: Exception): Crasher {
            logException(e)
            val path = MagmaLogger.flush(true)
            return Crasher(path)
        }
    }
}